src/libcrun/handlers/krun.c

```diff
@@ -293,6 +295,58 @@ libkrun_unload (void *cookie, libcrun_error_t *err arg_unused)
   return 0;
 }
 
+static runtime_spec_schema_defs_linux_device_cgroup *
+make_oci_spec_dev (const char *type, dev_t device, bool allow, const char *access)
+{
+  runtime_spec_schema_defs_linux_device_cgroup *dev = xmalloc0 (sizeof (*dev));
+
+  dev->allow = allow;
+  dev->allow_present = 1;
+
+  dev->type = xstrdup (type);
+
+  dev->major = major (device);
+  dev->major_present = 1;
+
+  dev->minor = minor (device);
+  dev->minor_present = 1;
+
+  dev->access = xstrdup (access);
+
+  return dev;
+}
+
+static int
+libkrun_modify_oci_configuration (void *cookie, libcrun_context_t *context,
+                                  runtime_spec_schema_config_schema *def,
+                                  libcrun_error_t *err)
+{
+  const size_t device_size = sizeof (runtime_spec_schema_defs_linux_device_cgroup);
+  struct stat st;
+  size_t len;
+  int ret;
+
+  if (def->linux == NULL || def->linux->resources == NULL
+      || def->linux->resources->devices == NULL)
+    return 0;
+
+  /* Always allow the /dev/kvm device.  */
+
+  ret = stat ("/dev/kvm", &st);
```

Comment:
I think we should do the same thing for `/dev/sev`, creating it if exists in the host and `handle_sev != NULL`. Then we can drop the `HANDLER_CONFIGURE_AFTER_MOUNTS` code in `libkrun_configure_container()`, simplifying the logic behind the creation of those devices.

Reply:
I've not touched that as I've no way to test it.  I can copy the same logic as `/dev/kvm`, it should be ok but I cannot guarantee that 100% :)

Reply:
I've added two new patches but I could not verify it doesn't break `/dev/sev`

Reply:
Great, thanks! Let me give it a quick try on an SEV machine.